python/lsst/analysis/drp/skyPlot.py

```diff
@@ -0,0 +1,351 @@
+# recommended syntax
+import pandas as pd
+pd.options.mode.chained_assignment = None  # default='warn'
+
+
+class SkyPlotTaskConnections(pipeBase.PipelineTaskConnections, dimensions=("tract", "skymap"),
+                             defaultTemplates={"inputCoaddName": "deep", "plotName": "deltaCoords",
+                                               "tableType": "forced"}):
+
+    catPlot = pipeBase.connectionTypes.Input(doc="The tract wide catalog to make plots from.",
+                                             storageClass="DataFrame",
+                                             name="qaTractTable_{tableType}",
+                                             dimensions=("tract", "skymap", "band"))
+
+    skymap = pipeBase.connectionTypes.Input(doc="The skymap for the tract",
+                                            storageClass="SkyMap",
+                                            name=BaseSkyMap.SKYMAP_DATASET_TYPE_NAME,
+                                            dimensions=("skymap",))
+
+    skyPlot = pipeBase.connectionTypes.Output(doc="A plot showing the on sky distribution of a value.",
+                                              storageClass="Plot",
+                                              name="skyPlot_{plotName}",
+                                              dimensions=("tract", "skymap"))
+
+
+class SkyPlotTaskConfig(pipeBase.PipelineTaskConfig, pipelineConnections=SkyPlotTaskConnections):
+
+    # These can stop being functors as defaults if the catalogue
+    # moves natively to degrees
+    xColName = pexConfig.Field(
+        doc="The column name for the values to be plotted on the x axis.",
+        dtype=str,
+        default="Functor",
+    )
+
+    yColName = pexConfig.Field(
+        doc="The column name for the values to be plotted on the y axis.",
+        dtype=str,
+        default="Functor",
+    )
+
+    xLabel = pexConfig.Field(
+        doc="The x axis label",
+        dtype=str,
+        default="Right Ascension (deg)",
+    )
+
+    yLabel = pexConfig.Field(
+        doc="The y axis label",
+        dtype=str,
+        default="Declination (deg)",
+    )
+
+    sourceTypeColName = pexConfig.Field(
+        doc="The column to use for star - galaxy separation.",
+        dtype=str,
+        default="base_ClassificationExtendedness_flag",
+    )
+
+    objectsToPlot = pexConfig.Field(
+        doc="Which types of objects to include on the plot, should be one of 'stars', 'galaxies' or 'all'.",
+        dtype=str,
+        default="stars",
+    )
+
+    zColName = pexConfig.Field(
+        doc="Which column to use to color code the points.",
+        dtype=str,
+        default="Functor"
+    )
+
+    axisFunctors = ConfigStructField(
+        doc="The functor to use to calculate the values used on each axis. Used if <axis>ColName is "
+            "set to 'Functor'.",
+        default={"xFunctor": RadToDeg, "yFunctor": RadToDeg, "zFunctor": SNCalculator},
```

Comment:
The x and y columns are configurable to allow for users to make changes to the column if they wish, e.g. being in radians or degrees. It will also make it easier for this plot to be used to make generic 3 axis plots. While this is currently not quite feasible as adding the patches in the background is hardcoded it will probably be done on a future ticket.